pkg/auth/local_authenticator.go

```diff
@@ -0,0 +1,90 @@
+package auth
+
+import (
+	"crypto"
+	"net/http"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/runtime/security"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/assisted-service/internal/common"
+	logutil "github.com/openshift/assisted-service/pkg/log"
+	"github.com/openshift/assisted-service/pkg/ocm"
+)
+
+type LocalAuthenticator struct {
+	log       logrus.FieldLogger
+	publicKey crypto.PublicKey
+}
+
+func NewLocalAuthenticator(cfg *Config, log logrus.FieldLogger) (*LocalAuthenticator, error) {
+	if cfg.ECPublicKey == "" {
+		return nil, errors.Errorf("local authentication requires an ecdsa Public Key")
+	}
+
+	key, err := jwt.ParseECPublicKeyFromPEM([]byte(cfg.ECPublicKey))
+	if err != nil {
+		return nil, err
```

Comment:
Shouldn't error handling behave the same in NewRHSSOAuthenticator:
https://github.com/openshift/assisted-service/blob/master/pkg/auth/rhsso_authenticator.go#L44
I mean, any reason it throws the error here, whereas hides the error in rhsso?

Reply:
I actually had changed that in a previous version of this.

IMO we shouldn't ever exit the program from a package other than `main`.
The code in main can decide how it wants to deal with failing to initialize the authenticator.

Reply:
And it is actually already doing that:

https://github.com/openshift/assisted-service/blob/5c64d7b843e9d024826d172cfd02545b8a889813/cmd/main.go#L187-L188

Reply:
> I actually had changed that in a previous version of this.
> 
> IMO we shouldn't ever exit the program from a package other than `main`.
> The code in main can decide how it wants to deal with failing to initialize the authenticator.

+1

Reply:
right, I've actually just wondered why we don't return the error in NewRHSSOAuthenticator ctr.
but not relevant for this pr :)